utils/zerovec/src/map/hashmap.rs

```diff
@@ -0,0 +1,270 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use super::*;
+use crate::ule::AsULE;
+use crate::ZeroVec;
+use ahash::AHasher;
+use alloc::borrow::Borrow;
+use alloc::vec::{from_elem, Vec};
+use core::hash::{Hash, Hasher};
+
+#[inline]
+fn create_hasher_with_seed(seed: u128) -> AHasher {
+    AHasher::new_with_keys(seed, 0xaabbccdd)
+}
+
+#[inline]
+fn compute_hash<K: Hash>(seed: u32, k: K, m: usize) -> u32 {
+    let mut hasher = create_hasher_with_seed(seed.into());
+    k.hash(&mut hasher);
+    (hasher.finish() % m as u64) as u32
```

Comment:
```suggestion
    (hasher.finish() as usize % m) as u32
```

This avoids doing a `u64` mod instruction on 32-bit architectures.

Reply:
Done.

Reply:
I have to take this back, this does not give the same results on 32 and 64 bit. Assume the hash is 2^32 + 1, then on 32 bit that gets truncated to 1, which is not congruent to 2^32+1 under arbitrary moduli.

I don't think we need 64 bits here, so let's do the arithmetic in 32 bits and then widen to usize.